Core/Source/Observable/Observable.swift

```diff
@@ -68,27 +68,32 @@ public final class ObserverList<Event>: GenericObservable<Event> {
     }
 
     public override func addObserver(_ observer: @escaping (Event) -> Void) -> ObserverToken {
-        precondition(Thread.isMainThread, "addObserver must run on main thread")
         let token = Token.makeUnique()
-        self.observers[token] = observer
+        locked { self.observers[token] = observer }
         return token
     }
 
     public override func removeObserver(with token: ObserverToken) {
-        precondition(Thread.isMainThread, "removeObserver must run on main thread")
-        self.observers[token] = nil
+        locked { self.observers[token] = nil }
     }
 
     public func notify(_ event: Event) {
         precondition(Thread.isMainThread, "notify must run on main thread")
         /// Create a local array in case notification causes an observer to be removed.
-        let observerValues = self.observers.values
+        let observerValues = locked { self.observers.values }
         for observer in observerValues {
             observer(event)
         }
     }
 
-    fileprivate var observers: [ObserverToken: (Event) -> Void]
+    private var observers: [ObserverToken: (Event) -> Void]
+    private var lock = pthread_mutex_t()
```

Comment:
pthread mutexes have to be initialized using pthread_mutex_init otherwise they don't do anything 🍾  because the default initializer basically just creates a empty value. You also should call pthread_mutex_destroy in the deinit. 

Given how easy it is to screw up, whenever I don't need a recursive lock I prefer just using a serial DispatchQueue. Since the `sync` function (somewhat recently) returns the value from the closure these calls can remain one-liners:

```
private let subscriptionQueue = DispatchQueue()
```
```
let observerValues = subscriptionQueue.sync { self.observers.values }
```